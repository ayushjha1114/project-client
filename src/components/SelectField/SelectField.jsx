import React from 'react';
import PropTypes from 'prop-types';
import style from './style';

const propTypes = {
  error: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  options: PropTypes.arrayOf(PropTypes.object),
  defaultText: PropTypes.string,
};

// default values for props:
const defaultProps = {
  value: '',
  error: '',
  options: [],
  defaultText: 'Select',
  onChange: () => {},
};

class SelectField extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const {
      error,
      defaultText,
      options,
      ...rest
    } = this.props;
    return (
      <>
        <div>
          <select {...rest} style={{ ...style.base }}>
            <option key={defaultText} value={defaultText}>{defaultText}</option>
            {
              options.map(item => (
                <option key={item.label} value={item.label}>{item.label}</option>
              ))
            }
          </select>
          { (error) ? <p style={{ ...style.para }}>{error}</p> : '' }
        </div>
      </>
    );
  }
}
SelectField.propTypes = propTypes;
SelectField.defaultProps = defaultProps;

export default SelectField;
