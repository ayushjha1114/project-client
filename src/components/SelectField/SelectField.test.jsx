import React from 'react';
import { shallow, mount } from 'enzyme';
import SelectField from './SelectField';

describe('SelectField', () => {
  test('should render correctly', () => {
    const component = shallow(<SelectField />);
    expect(component).toMatchSnapshot();
  });
  it('check options props', () => {
    const props = {
      options: ['asasa'],
    };

    const component = mount(<SelectField {...props} />);
    expect(component.prop('options')).toEqual(['asasa']);
  });
});
