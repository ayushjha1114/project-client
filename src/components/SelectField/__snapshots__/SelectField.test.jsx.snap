// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectField
    defaultText="Select"
    error=""
    onChange={[Function]}
    options={Array []}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div>
        <select
          onChange={[Function]}
          style={
            Object {
              "padding": "10px",
              "width": "100%",
            }
          }
          value=""
        >
          <option
            value="Select"
          >
            Select
          </option>
        </select>
        
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <select
            onChange={[Function]}
            style={
              Object {
                "padding": "10px",
                "width": "100%",
              }
            }
            value=""
          >
            <option
              value="Select"
            >
              Select
            </option>
          </select>,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="Select"
              >
                Select
              </option>,
              Array [],
            ],
            "onChange": [Function],
            "style": Object {
              "padding": "10px",
              "width": "100%",
            },
            "value": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Select",
              "nodeType": "host",
              "props": Object {
                "children": "Select",
                "value": "Select",
              },
              "ref": null,
              "rendered": "Select",
              "type": "option",
            },
          ],
          "type": "select",
        },
        "",
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div>
          <select
            onChange={[Function]}
            style={
              Object {
                "padding": "10px",
                "width": "100%",
              }
            }
            value=""
          >
            <option
              value="Select"
            >
              Select
            </option>
          </select>
          
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              onChange={[Function]}
              style={
                Object {
                  "padding": "10px",
                  "width": "100%",
                }
              }
              value=""
            >
              <option
                value="Select"
              >
                Select
              </option>
            </select>,
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="Select"
                >
                  Select
                </option>,
                Array [],
              ],
              "onChange": [Function],
              "style": Object {
                "padding": "10px",
                "width": "100%",
              },
              "value": "",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "Select",
                "nodeType": "host",
                "props": Object {
                  "children": "Select",
                  "value": "Select",
                },
                "ref": null,
                "rendered": "Select",
                "type": "option",
              },
            ],
            "type": "select",
          },
          "",
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
